systemtest/src/test/java/io/strimzi/systemtest/migration/MigrationST.java

```diff
@@ -239,17 +283,64 @@ void testMigrationFromZkToKRaft(ExtensionContext extensionContext) {
+        ClientUtils.waitForClientsSuccess(continuousClients.getProducerName(), continuousClients.getConsumerName(), testStorage.getNamespaceName(), continuousClients.getMessageCount());
+    }
+
+    private void doRollback(TestStorage testStorage) {
+        LOGGER.info("From {} state we are going to roll back to ZK", KafkaMetadataState.KRaftPostMigration.name());
+
+        LOGGER.info("Rolling migration back - applying {}: rollback annotation", Annotations.ANNO_STRIMZI_IO_KRAFT);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getMetadata().getAnnotations().put(Annotations.ANNO_STRIMZI_IO_KRAFT, "rollback"), testStorage.getNamespaceName());
 
-        LOGGER.info("13a. checking if metadata about KafkaTopic: {} are in KRaft controller", kraftTopicName);
+        LOGGER.info("Waiting for Broker Pods to be rolled - bringing back dependency on ZK");
+        brokerPodsSnapshot = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), brokerSelector, 3, brokerPodsSnapshot);
 
-        assertThatTopicIsPresentInKRaftMetadata(controllerSelector, kraftTopicName);
+        LOGGER.info("Waiting until .status.kafkaMetadataState contains {} state", KafkaMetadataState.KRaftDualWriting.name());
+        KafkaUtils.waitUntilKafkaStatusContainsKafkaMetadataState(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaMetadataState.KRaftDualWriting.name());
 
-        LOGGER.info("13b. checking if we are able to do a message transmission on KafkaTopic: {}", kraftTopicName);
+        LOGGER.info("Deleting Controller's NodePool");
+        KafkaNodePool controllerPool = KafkaNodePoolResource.kafkaNodePoolClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getControllerPoolName()).get();
+        resourceManager.deleteResource(controllerPool);
+
+        LOGGER.info("Applying {}: disabled annotation", Annotations.ANNO_STRIMZI_IO_KRAFT);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getMetadata().getAnnotations().put(Annotations.ANNO_STRIMZI_IO_KRAFT, "disabled"), testStorage.getNamespaceName());
+
+        LOGGER.info("Waiting for Broker Pods to be rolled - rolling back from DualWrite mode to Zookeeper");
+        brokerPodsSnapshot = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), brokerSelector, 3, brokerPodsSnapshot);
+
+        LOGGER.info("Waiting until .status.kafkaMetadataState contains {} state", KafkaMetadataState.ZooKeeper.name());
+        KafkaUtils.waitUntilKafkaStatusContainsKafkaMetadataState(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaMetadataState.ZooKeeper.name());
+
+        LOGGER.info("Checking that __cluster_metadata topic does not exist in Kafka Brokers");
+        assertThatClusterMetadataTopicNotPresentInBrokerPod(testStorage.getNamespaceName(), brokerSelector);
```

Comment:
while it's important to check this here, what do you think about checking the opposite (that the __cluster_metadata topic exists) but at the beginning of the test, I mean in post-migration before running the rollback?

Reply:
Using that file in the Kafka folder we are checking that the newly created topic is there (that's the first step of the migration -> when the metadata state is in KRaftPostMigration). That means that we _know_ that the file is there. But if you think that it will be worth adding it, I will do that -> we are anyway checking the presence of the topic inside the file in controller Pod

Reply:
added the check at the beginning of the `doRollback()` method